plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,411 @@
+                    tensorArray.add(numberList);
+                }
+                modelOutputValue = tensorArray;
+                if (modelOutputValue == null) {
+                    throw new RuntimeException("model inference output cannot be null");
+                }
+            }
+
+            List<String> dotPathsInArray = writeNewDotPathForNestedObject(ingestDocument.getSourceAndMetadata(), newDocumentFieldName);
+
+            if (dotPathsInArray.size() == 1) {
+                ValueSource ingestValue = ValueSource.wrap(modelOutputValue, scriptService);
+                TemplateScript.Factory ingestField = ConfigurationUtils
+                    .compileTemplate(TYPE, tag, newDocumentFieldName, newDocumentFieldName, scriptService);
+                ingestDocument.appendFieldValue(ingestField, ingestValue, false);
+            } else {
+                ArrayList<?> modelOutputValueArray;
+                if (modelOutputValue instanceof List) {
+                    List<?> modelOutputList = (List<?>) modelOutputValue;
+                    modelOutputValueArray = new ArrayList<>(modelOutputList);
+                } else if (modelOutputValue instanceof ArrayList) {
+                    modelOutputValueArray = (ArrayList<?>) modelOutputValue;
+                } else {
+                    throw new IllegalArgumentException("model output is not an array, cannot assign to array in documents.");
+                }
+
+                // check length of the prediction array to be the same of the document array
+                if (dotPathsInArray.size() != modelOutputValueArray.size()) {
+                    throw new RuntimeException(
+                        "the prediction field: "
+                            + originalModelOutputFieldName
+                            + " is an array in size of "
+                            + modelOutputValueArray.size()
+                            + " but the document field array from field "
+                            + newDocumentFieldName
+                            + " is in size of "
+                            + dotPathsInArray.size()
+                    );
+                }
+                // Iterate over dotPathInArray
+                for (int i = 0; i < dotPathsInArray.size(); i++) {
+                    String dotPathInArray = dotPathsInArray.get(i);
+                    Object modelOutputValueInArray = modelOutputValueArray.get(i);
+                    ValueSource ingestValue = ValueSource.wrap(modelOutputValueInArray, scriptService);
+                    TemplateScript.Factory ingestField = ConfigurationUtils
+                        .compileTemplate(TYPE, tag, dotPathInArray, dotPathInArray, scriptService);
+                    ingestDocument.appendFieldValue(ingestField, ingestValue, false);
+                }
+            }
+        } else {
+            throw new RuntimeException("model inference output cannot be null");
```

Comment:
This will return 500 error, is that expected ?

Reply:
when the model output return null, it return the exception to the output. Added a ignoreFailure flag, so if users would like to ignoreFailure and continue ingestDocument, it can turn on the ignoreFailure to true. 

This is an example of test case, 
https://github.com/opensearch-project/ml-commons/blob/b9dfc86ad0d656530d1f112d11704828691bd335/plugin/src/test/java/org/opensearch/ml/processor/MLInferenceIngestProcessorTests.java#L836